### DISCUSSION TITLE
Livewire with MediaLibrary (not Pro)

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#2389] [1. Help] [answered]
I've got a component that stores the user details into the database. Part of this is storing the profile picture in Spatie's MediaLibrary. I know MediaLibrary Pro supports Livewire, we do have that license too, but I'd like to build this component without it, to have a little more flexibility in terms of how it works + design.

In our `save()` method we have this:

```php
if (! is_null($this->profile)) {
    $this->user
        ->addMedia($this->profile->getRealPath())
        ->usingName($this->profile->getClientOriginalName())
        ->toMediaCollection('profile');
}
```

When we show the image on the page, inside an if-statement that checks if $profile is set, with:

```blade
<img src="{{ $profile->temporaryUrl() }}" alt="Profile Photo">
```

It shows the image just fine.

But when I hit save it errors out with:

```
File not found at path: livewire-tmp/5oaNzbvdAh2eRjqGJ9wHjQxujioe1M-metaUGhvdG8gMi5qcGc=-.jpg
```

To debug it, this is what I did:

- The path `storage/app/livewire-tmp` is empty before I start.
- I then select an image, it shows on the page.
- I hit submit, the path is `$this->profile->getRealPath()` is the fill path with `livewire-tmp/5oaNzbvdAh2eRjqGJ9wHjQxujioe1M-metaUGhvdG8gMi5qcGc=-.jpg` at the end of it.
- Using `ls` it shows the file is in `storage/app/livewire-tmp`.
- It then calls `$this->user->addMedia()`. This fires the error.
- After the error is called, the file in `storage/app/livewire-tmp` is gone.

To be 1000% sure I'm not getting lost between requests I used `ray()` to output the path and `ray()->pause()` just before the `addMedia()` call runs.

I'm in doubt if this is an issue with Livewire, MediaLibrary or my own code. If I have to assume, it would be my code :-)

**Comment by ju5t** [comment, marked as answer]
...

I just typed a massive reply with all the addition research I did. Turns out that the error isn't caused by any of that. The error comes from an if-statement in the blade view.

```blade
@if ($avatar)
    <img src="{{ $profile->temporaryUrl() }}" alt="Profile Photo">
@else
    <img src="{{ $profile->avatarUrl() }}" alt="Profile Photo">
@endif
```

Apparently, it's trying to do a mime check on the `temporaryUrl` method. This fails because MediaLibrary moved the file when adding it to the user. This makes sense.

What doesn't make sense is why it tries to open the file to begin with. I guess it might have something to do with how file uploads work in Livewire as it uses emit (if I remember correctly).

For now my issue is solved by adding a `$profile->exists()` check to the if statement, I will come back later on this.

**Reply by MohamedMaher5** [reply]
@crayon1337
